tests/data/components/Button/index.tsx

```diff
@@ -75,15 +86,24 @@ const Dot = styled.div`
 	border-radius: 50%;
 	margin-right: ${({ dotCount, index, dotSpacing }) =>
 		`${index < dotCount - 1 ? dotSpacing : 0}px`};
-	animation: ${({ totalInterval, blinkInterval, index }) =>
-		`${totalInterval}s ${(blinkInterval / 2) *
+	animation: ${({ dotCount, blinkInterval, index }) =>
+		`${dotCount * blinkInterval}s ${(blinkInterval / 2) *
 			index}s ${blink} step-end infinite`};
 `;
 
 /**
  * Button Description
  */
-class Button extends Component {
+class Button extends Component<{
```

Comment:
This PR is standard setting so I wonder if we should experiment a bit here. I don't particularly like the in line type argument, but I don't know if react allows us to use an interface or declared type here. Thoughts on other ways of declaring the types?

Reply:
Pulled them out into interfaces